### DISCUSSION TITLE
Encryption?

### DISCUSSION THREAD

**Discussion by jaredteladoc** [mergesort/Boutique#7] [General]
This looks awesome...amazing work!  Any thought of adding ability to encrypt data?  I'd love to use this but data encryption important for my projects.  Thanks!

**Comment by mergesort** [comment]
Hey @jaredteladoc, thanks for the question and suggestion! I hadn't considered encryption so can you tell me more about your use case?

Under the hood Boutique saving files to disk using [Bodega](https://github.com/mergesort/bodega) so I think it should be possible to encrypt the data you want to store on disk (using the `add(_ item:)` or `add(_ items:)`) method and decrypt the data when you want to read it, but I want to make sure I can provide a good suggestion or brainstorm ideas if it's not immediately feasible.

**Comment by samalone** [comment]
@jaredteladoc, does this need to be part of Boutique, or can you put your encryption/decryption logic in the Codable functions of your class?

What I'm thinking is something like this:

1. Write EncryptedEncoder and DecryptedDecoder adapter classes that perform the encryption.
2. In the `init(from: Decoder)` and `encode(to: Encoder)` methods of your class, wrap your encrypted Encoder/Decoder objects around the provided encoder/decoder and read/write your sensitive data to those.

This isn't quite as convenient as having encryption built-in, but it keeps Boutique simple and gives you a lot of control over which data is encrypted, the algorithms used, and where the keys come from.

Maybe the EncryptedEncoder and DecryptedDecoder classes could be part of a companion library to Boutique?
